### ISSUE TITLE
Regression: fetch requests in middleware fail with a body

### ISSUE THREAD

**Issue by ottomated** [vercel/next.js#38963]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP PREEMPT_DYNAMIC Fri, 15 Jul 2022 15:33:02 +0000
    Binaries:
      Node: 16.15.0
      npm: 8.5.5
      Yarn: 3.2.1
      pnpm: N/A
    Relevant packages:
      next: 12.2.4-canary.0
      eslint-config-next: 12.2.3
      react: 18.2.0
      react-dom: 18.2.0
```

### Describe the Bug

Making a `fetch` request from middleware that includes a request body fails with `TypeError: fetch failed`.

Tested in `12.2.2` - succeeds. Fails in `12.2.3` and `canary`.

### Expected Behavior

The fetch request should succeed.

### Link to reproduction

https://codesandbox.io/s/jovial-fast-yflq3w?file=/middleware.ts

### To Reproduce

1. Load any page
2. Observe the error in the console
3. Remove the line `body: ""`
4. Observe that there is no error

**Comment by joaogarin** [comment]
Maybe take a look in https://nextjs.org/docs/messages/middleware-upgrade-guide#no-response-body seems like this has changed in 12.2 so the behaviour is intentional : 

"If your Middleware does respond with a body, a runtime error will be thrown"

**Comment by balazsorban44** [comment]
@joaogarin that only applies to responding (ie: `return`) from the Middleware. Using `fetch` should otherwise work as expected.

This seems to be related to #38862, thanks for reporting.

**Comment by Charlie-007-star** [comment]
is there a solution for that yet?

**Comment by dferber90** [comment]
@Charlie-007-star Yes, see https://github.com/vercel/edge-runtime/pull/68

**Comment by balazsorban44** [comment]
#39219 included the fix for this and will be available in the next `next@canary` release.

**Comment by balazsorban44** [comment]
This should be out on `12.2.4-canary.11`, please check it out!

**Comment by trm217** [comment]
@balazsorban44 This error started occurring in a project using next@12.2.3.
Unfortunately it was not fixed with next@12.2.4.
In [this file](https://github.com/republik/plattform/blob/main/apps/www/lib/helpers/middleware/FetchMeObject.ts) you can see the query that causes the issues for us.
